frontend/src/metabase/lib/click-behavior.js

```diff
@@ -226,3 +232,60 @@ export function hasActionsMenu(dashcard) {
 export function isTableDisplay(dashcard) {
   return dashcard.card.display === "table";
 }
+
+export function formatSourceForTarget(
+  source,
+  target,
+  { data, extraData, clickBehavior },
+) {
+  const datum = data[source.type][source.id.toLowerCase()] || [];
+  if (datum.column && isDate(datum.column)) {
+    if (target.type === "parameter") {
+      // we should serialize differently based on the target parameter type
+      const parameter = getParameter(target, { extraData, clickBehavior });
+      if (parameter) {
+        return formatDateForParameterType(datum.value, parameter.type);
+      }
+    } else {
+      // If the target is a dimension or variable,, we serialize as a date to remove the timestamp.
+      // TODO: provide better serialization for field filter widget types
+      return formatDateForParameterType(datum.value, "date/single");
+    }
+  }
+  return datum.value;
+}
+
+function formatDateForParameterType(value, parameterType) {
+  const m = parseTimestamp(value);
+  if (!m.isValid()) {
+    return String(value);
+  }
+  if (parameterType === "date/month-year") {
```

Comment:
do we have these values as `const`s anywhere to refer to? I tend to prefer that over strings because typos and such

Reply:
We don't currently, but you're right that we should! Static verification that there are no typos is definitely worth typing out another `import`.